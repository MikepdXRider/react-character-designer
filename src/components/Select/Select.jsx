import React from 'react'

export default function Select({optionsArr, setState, state}) {
    // console.log('setState: ', setState);
    // console.log('optionsArr: ', optionsArr);
    return (
        <select value={state} onChange={(e) => setState(e.target.value)}>
            <option value="" disabled hidden>Please Choose...</option>
            {
                optionsArr.map(option => <option key={option} value={option} data-testid={option}>{option}</option>)
            }
        </select>
    )
}
