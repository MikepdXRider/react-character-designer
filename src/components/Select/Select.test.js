import {screen, render} from '@testing-library/react'
import Select from './Select.jsx'

it('renders select component', () => {
    // render enables us to pseudo-render a component. This also lets us use the screen method later on
    const {container} = render(<Select optionsArr={['test1', 'test2', 'test3']} />);
    //data-testid attributes hardcoded onto jsx tags act as ids that can be grabbed here.
    const optionEl = screen.getByTestId('test1');
    // expect the above elements to be in the rendered document.
    expect(optionEl).toBeInTheDocument();
    // expect container to match the produced snapshot(first test generates, all other test compare)
    expect(container).toMatchSnapshot();
})