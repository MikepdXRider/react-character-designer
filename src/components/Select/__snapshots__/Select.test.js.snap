// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders select component 1`] = `
<div>
  <select>
    <option
      disabled=""
      hidden=""
      value=""
    >
      Please Choose...
    </option>
    <option
      data-testid="test1"
      value="test1"
    >
      test1
    </option>
    <option
      data-testid="test2"
      value="test2"
    >
      test2
    </option>
    <option
      data-testid="test3"
      value="test3"
    >
      test3
    </option>
  </select>
</div>
`;
